### PR TITLE
Add deps to makefile and travis yaml file

### PR DESCRIPTION
**Pull request by rydrman**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+go:
+    - tip
+install:
+    make deps
+script:
+    make test
+notifications:
+  slack: 
+    rooms:
+      - kloudktrl:741EF7gPvqNNBF4U8nW4ObDg#notifications
```
Comment (seanhagen) [inline .travis.yml]: Is there no way to have this stored in an encrypted file? If nobody else can use this to send notifications to our slack channel that's fine, I just don't want to open ourselves up to spamming.

File: Makefile [hunk 2]
```diff
@@ -22,6 +22,11 @@ BUILD_TIME=`date +%FT%T%z`
 REPO=github.com/ktrl.io/upload
 LDFLAGS=-ldflags "-X ${REPO}/core.Version=${VERSION} -X ${REPO}/core.BuildTime=${BUILD_TIME}"
 
+GODEPS=$(subst [,, $(subst ],, $(shell go list -f '{{.Deps}}')))
```
Comment (seanhagen) [inline Makefile]: How/where is the `.Deps` folder (...file?) generated? I don't see one in KloudKtrl/auth-service. Is it a `godep` thing?

File: Makefile [hunk 3]
```diff
@@ -42,7 +47,10 @@ vet:
 	go vet
 
 deps:
-	# godep go install
+	@for dep in $(GODEPS); do \
+		echo go get $$dep; \
+		go get $$dep; \
+	done
```
Comment (seanhagen) [inline Makefile]: Love this bit -- not having to keep track of all the deps by hand is great ( ...I may have been doing that in previous versions of the `deps` target )

**Comment by seanhagen** [comment]
Looks good, just worried about the slack key in `travis.yml`. If we don't have to worry about that key being public, groovy.

**Comment by rydrman** [comment]
Ya I didn't even think of that, looks like it can be encrypted, I'll do that tomorrow and update

File: Makefile (hunk 2 above)
Comment (rydrman) [inline Makefile]: No folder is just output of the go list command, seems to access the internal go dependency list for the project

File: .travis.yml [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
-script:
-    make test
+- tip
+install: make deps
+script: make test
 notifications:
-  slack: 
+  slack:
     rooms:
-      - kloudktrl:741EF7gPvqNNBF4U8nW4ObDg#notifications
```
Comment (seanhagen) [inline .travis.yml]: Is the encrypted token based on this key, or did you generate a new key to use? Now that this key is in the history of the repo, I'd like to make sure it's useless.

**Comment by seanhagen** [comment]
Looks like coveralls does run, it just has to wait for travis to finish. Looks good though, merging.

**Pull request merged**